internal/run/convert.go

```
@@ -0,0 +1,190 @@
+	releaseVersionsMax int
+	tillerNS string
+
+	cmd cmd
+}
+
+func NewConvert(cfg env.Config, tillerNS string, kubeConfig string) *Convert {
+	return &Convert{
+		config: newConfig(cfg),
+		release: cfg.Release,
+		deleteV2Releases: cfg.DeleteV2Releases,
+		dryRun: cfg.DryRun,
+		kubeContext: cfg.Context,
+		kubeConfig: kubeConfig,
+		label: cfg.TillerLabel,
+		releaseVersionsMax: cfg.ReleaseVersionsMax,
+		tillerNS: tillerNS,
+	}
+}
+
+// Execute runs the `2to3 convert` command from Prepare
+// it detects if a conversion is needed and if not it will return no error
+// if a conversion is required then any returned error is related to a step
+// in the conversion process
+func (c *Convert) Execute() error {
+
+	clientSet, err := clientSetFromKubeConfig(c.kubeConfig)
+	if err != nil {
+		return err
+	}
+
+	// If Tiller is not present, utils.ListReleasesWithKubeConfig will panic
+	if !isTillerPresent(clientSet, "name=tiller,app=helm") {
+		log.Println("There is no Tiller Deployment running")
+		return nil
+	}
+
+	// Check for existence of v2 Release (only if Tiller is running)
+	var v2ReleaseFound bool
+
+	listOptions := utils.ListOptions{
+		ReleaseName: c.release,
+		TillerNamespace: c.tillerNS,
+		TillerLabel: c.label,
+	}
+
+	v2Releases, _ := utils.ListReleasesWithKubeConfig(listOptions, c.kubeConfig, c.kubeContext)
+	if len(v2Releases) > 0 {
+		log.Printf("A v2 Release of %s was found", c.release)
+		v2ReleaseFound = true
+	} else {
+		log.Printf("There is no v2 Release of %s to migrate", c.release)
+		return nil
+	}
+
+	// Check for existence of v3 Release
+	var v3ReleaseFound bool
+
+	kc := common.KubeConfig{
+		Context: c.kubeContext,
+		File:    c.kubeConfig,
+	}
+
+	v3cfg, err := v3.GetActionConfig(c.namespace, kc)
+	if err != nil {
+		return err
+	}
+
+	_, err = v3cfg.Releases.Deployed(c.release)
+	if err != nil {
+		log.Printf("No v3 Release of %s found", c.release)
+	} else {
+		log.Printf("A v3 Release of %s was found", c.release)
+		v3ReleaseFound = true
```

This variable doesn't look necessary as you can simply return the following error in this else block.